kotlinx-coroutines-core/jvm/test/sync/SemaphoreStressTest.kt

```diff
@@ -2,10 +2,18 @@ package kotlinx.coroutines.sync
 
 import kotlinx.coroutines.*
 import org.junit.Test
+import org.junit.After
 import kotlin.test.assertEquals
 
 class SemaphoreStressTest : TestBase() {
 
+    private val pool = newSingleThreadContext("SemaphoreStressTest")
```

Comment:
Please, move `pool` into `stressReleaseCancellation()` test function, since other functions to do not need this single-thread context at the moment.